Collection.UnitTests/CollectonTests.cs

```diff
@@ -1,12 +1,282 @@
+
+using Collections;
+using NUnit.Framework.Constraints;
+
 namespace Collection.UnitTests
 {
+
     public class CollectionTests
     {
 
         [Test]
-        public void Test1()
+        public void Test_Collection_EmptyConstructor()
+        {
+            //Act & Arrange
+            var nums = new Collection<int>();
+
+            //Assert
+            Assert.That(nums.ToString(), Is.EqualTo("[]"));
+
+        }
+
+
+        [Test]
+        public void Test_Collection_ConstructorSingleItem()
+        {
+            var nums = new Collection<int>(5);
+
+            Assert.That(nums[0], Is.EqualTo(5));
+        }
+
+
+        [Test]
+        public void Test_Collection_ConstructorMultipleItems()
+        {
+            var nums = new Collection<int>(5, 10, 15);
+
+            Assert.That(nums.ToString(), Is.EqualTo("[5, 10, 15]"));
+        }
+
+
+        [Test]
+        public void Test_Collection_AddText()
+        {
+            var numbers = new Collection<string>("Zero", "One");
+
+            numbers.Add("Two");
+
+            Assert.That(numbers.ToString(), Is.EqualTo("[Zero, One, Two]"));
+        }
+
+
+        [Test]
+        public void Test_Collection_AddRangeWithGrow()
+        {
+            var nums = new Collection<int>();
+            int oldCapacity = nums.Capacity;
+            var newNums = Enumerable.Range(1000, 2000).ToArray();
+
+            nums.AddRange(newNums);
+            string expectedNums = "[" + string.Join(", ", newNums) + "]";
+
+            Assert.That(nums.ToString(), Is.EqualTo(expectedNums));
+            Assert.That(nums.Capacity, Is.GreaterThanOrEqualTo(oldCapacity));
+            Assert.That(nums.Capacity, Is.GreaterThanOrEqualTo(nums.Count));
+        }
+
+
+        [Test]
+        public void Test_Collection_GetByIndex()
+        {
+            // Arrange
+            var names = new Collection<string>("Peter", "Maria");
+            // Act
+            var item0 = names[0];
+            var item1 = names[1];
+            // Assert
+            Assert.That(item0, Is.EqualTo("Peter"));
+            Assert.That(item1, Is.EqualTo("Maria"));
+        }
+
+
+        [Test]
+        public void Test_Collection_GetByInvalidIndex()
```

Comment:
everything is correct in the test, but as a variant you can try to write them using DDT, e.g [TestCase(-1)]...